### PR TITLE
Enhancement/167/refactor entity templates

### PR DESCRIPTION
**Pull request by xcaverxiii**
complete issue #167 

Refactor some templates and create shared components

### CODE REVIEW

File: app/src/app/core/services/elasticsearch/data.service.ts [hunk 1]
```diff
@@ -183,4 +188,15 @@ export class DataService {
         }
         return entity;
     }
+
+    /**
+     * set type specific attributes
+     * @param entity entity object
+     */
+    private setTypes(entity: any) {
+        if (entity.type) {
```
Comment (mxxbe) [inline app/src/app/core/services/elasticsearch/data.service.ts]: The entity interface demands that type and id are set on entities - we use both here, but only check for the presence of one, so I would say it should either be a check for both or for none, for consistency.

File: app/src/app/shared/components/iconclass/iconclass.component.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<!-- Iconclass integration -->
+<div *ngIf="iconclassData" class="row">
```
Comment (mxxbe) [inline app/src/app/shared/components/iconclass/iconclass.component.html]: The iconclass label is now displayed in each case, but we want to hide it if the entity does not have iconclasses assigned to it. This probably got lost during the refactoring and should be fixed please.

image

Comment (xcaverxiii) [reply app/src/app/shared/components/iconclass/iconclass.component.html]: thx @mxxbe, i updated your conflicts.

File: app/src/app/core/services/elasticsearch/data.service.ts (hunk 1 above)
Comment (xcaverxiii) [reply app/src/app/core/services/elasticsearch/data.service.ts]: i added if entity.id condition

**Pull request merged**